src/Auxillary/json_schema/schema_generation.cpp

```diff
@@ -52,9 +52,12 @@ namespace Aux::schema {
     return 0;
   }
 
+  inline static json get_problem_data_schema();
+
   void make_schemas(Componentfactory const &factory, path schema_dir) {
 
     std::map<std::string, json> schemas{
+        {"problem_data", get_problem_data_schema()},
```

Comment:
the following lines, e.g.
```cpp
{"topology",
 factory.get_topology_schema(
             /*allow_required_defaults=*/false, /*include_external=*/false)},
```
should make sure that the "required" property is removed from the schemas - I am not sure why this is not working as intended. 
This function calls `component->get_topology_schema(allow_required_defaults)` defined in the templates in `Componentfactory.hpp` and these individual components call `remove_defaults_from_required` from `make_schema.cpp` if `allow_required_defaults` is false.

`remove_defaults_from_required` should remove all components with defaults.

Reply:
e.g. 

```cpp
if (opt_boundary_schema.has_value()) {
          auto boundary_schema = opt_boundary_schema.value();
          Aux::schema::add_defaults(
              boundary_schema,
              this->defaults.value("boundary_values", R"({})"_json));
          if (not allow_required_defaults) {
            Aux::schema::remove_defaults_from_required(boundary_schema);
          }
        }
```
`Aux::schema::add_defaults` is tested in `schemaCreationTest.cpp` so I am fairly certain this is working so I would probably set debug points in `remove_defaults_from_required`

Reply:
Does it work on your end?

Reply:
Fixed: You changed a copy of the required_list. I changed it to a reference. Will merge tomorrow.